pghoard/compressor.py

```diff
@@ -263,3 +282,82 @@ def set_state_defaults_for_site(self, site):
                     "count": 0
                 },
             }
+
+
+class WALFileDeleterThread(Thread):
+    """Deletes files which got compressed by the Compressor Thread, but keeps one file around
+
+    pg_receivewal, after some hickup, will use the files to compute the next wal segment to download.
+    if there are none, it will start at the current server position and so might miss wal segments. This
+    would mean that the backup is incomplete until the next base backup is taken and the service would
+    fail to be restored.
+
+    So the idea is to only unlink all files but the latest one.
+    """
+    def __init__(
+        self,
+        config: Dict,
+        wal_file_deletion_queue: Queue,
+        metrics: Metrics,
+    ):
+        super().__init__()
+        self.log = logging.getLogger("WALFileDeleter")
+        self.config = config
+        self.metrics = metrics
+        self.wal_file_deletion_queue = wal_file_deletion_queue
+        self.running = True
+        self.to_be_deleted_files: Dict[str, Set[str]] = {}
+        self.log.debug("WALFileDeleter initialized")
+
+    def run(self):
+        while self.running:
+            wait_timeout = 1.0
+            # config can be changed in another thread, so we have to lookup this within the loop
+            config_wait_timeout = self.config.get("deleter_event_get_timeout", wait_timeout)
+            try:
+                # prevent bad formats killing the thread
+                wait_timeout = float(config_wait_timeout)
+            except ValueError:
+                self.log.warning("Bad value for deleter_event_get_timeout: %r", config_wait_timeout)
+
```

Comment:
`float(None)` or `float({})` raise `TypeError`, not `ValueError`.
On the other hand, if the config value is a string that happens to be number-compatible, this converts the string, which isn't the expect behavior for the other config entries.

```suggestion
            import math
            if isinstance(config_wait_timeout, (float, int)) and math.isfinite(config_wait_timeout) and config_wait_timeout > 0:
                wait_timeout = config_wait_timeout
            else:
                self.log.warning("Bad value for deleter_event_get_timeout: %r, using default value instead: %r", config_wait_timeout, wait_timeout)

```

Reply:
Oh well, I wonder how many other places where we do nto do proper validation... We really should add pydantic to validate the config...